hibernate-rx-core/src/main/java/org/hibernate/rx/service/RxConnectionPoolProviderImpl.java

```diff
@@ -59,10 +62,21 @@ private Pool configurePool(Map configurationValues) {
 
 		final String url = ConfigurationHelper.getString(AvailableSettings.URL, configurationValues);
 		final URI uri = JdbcUrlParser.parse( url );
-		final String database = uri.getPath().substring( 1 );
+		String database = uri.getPath().substring( 1 );
+		if (uri.getScheme().equals("db2") && database.indexOf( ':' ) > 0) {
+			database = database.substring( 0, database.indexOf( ':' ) );
```

Comment:
Maybe you can add a comment to highlight how the db2 uri differs form the other ones